_posts/2016-03-28-Mutual-Problems.md

```diff
@@ -0,0 +1,130 @@
+---
+layout: post
+name: mutual-problems
+title: Mutual Problems
+date: 2016-03-28 00:00:00 +00:00
+author: Robert Firek
+image:
+    src: /assets/img/custom/blog/2016-03-28-https.jpg
+    attribution:
+        text: Image by Sean MacEntee (CC BY 2.0)
+        href: https://www.flickr.com/photos/smemon/15944989872/in/photostream/
+tags:
+- Java
+- HTTPS
+- Mutual Authentication
+--- 
+
+The HTTPS protocol is the well-established standard for securing our internet connections. Understanding how this protocol works is not problem and the corresponding [RFC document](https://tools.ietf.org/html/rfc2818) is available since 2000.
+
+Despite HTTPS is used so widely, you can still find a software which doesn't handle this protocol without unnecessary complexity. Unfortunately I'v experienced  problems during the implementation of [the mutual authentication](https://en.wikipedia.org/wiki/Mutual_authentication) in the language which should not surprise me at all. It is **Java**.
```

Comment:
I'm not sure to understand the part about you not being surprised. Are you saying you should not be surprised because Java always has unnecessary complexity? Or are you saying that you know Java well and you shouldn't have surprises when implementing client authentication in Java?